src/test/java/software/amazon/smithy/lsp/document/DocumentParserTest.java

```diff
@@ -319,4 +328,63 @@ enum Baz {
         assertThat(getInputA.kind(), equalTo(DocumentShape.Kind.DefinedMember));
         assertThat(getInputA.shapeName(), string("a"));
     }
+
+    @ParameterizedTest
+    @MethodSource("contiguousRangeTestCases")
+    public void findsContiguousRange(SourceLocation input, Range expected)  {
+        String text = """
+            abc def
+              ghi jkl
+            mno  pqr
+            """;
+        DocumentParser parser = DocumentParser.of(safeString(text));
+
+        Range actual = parser.findContiguousRange(input);
+
+        if (expected == null) {
+            assertNull(actual);
+        } else {
+            assertEquals(expected, actual);
+        }
+    }
+
+    private static Stream<Arguments> contiguousRangeTestCases() {
```

Comment:
Can you do two more tests - one at the very start and one at the very end of the file? I've been doing that as a sanity check since these type of index-offset calculations can be error-prone

Reply:
Yes. I can.

Reply:
Added.